classes/order/Order.php

```diff
@@ -1080,20 +1080,45 @@ public static function getCustomerNbOrders($id_customer)
     }
 
     /**
-     * Get an order by its cart id
+     * Get an order id by its cart id
      *
      * @param int $id_cart Cart id
-     * @return array Order details
+     * @return int Order id
+     * @deprecated since 1.7.1.0 Use getIdByCartId() instead
      */
     public static function getOrderByCartId($id_cart)
     {
-        $sql = 'SELECT `id_order`
-                FROM `'._DB_PREFIX_.'orders`
-                WHERE `id_cart` = '.(int)$id_cart
-                    .Shop::addSqlRestriction();
-        $result = Db::getInstance()->getRow($sql);
+        return self::getIdByCartId($id_cart);
+    }
+
+    /**
+     * Get an order object by its cart id
+     *
+     * @param int $id_cart Cart id
+     * @return OrderCore
+     */
+    public static function getByCartId($id_cart)
+    {
+        $id_order = self::getIdByCartId($id_cart);
+        $order = new self($id_order);
+
+        return (isset($order->id)) ? $order : null;
```

Comment:
Isn't it better to make something like that, here: 

$id_order = (int)self::getIdByCartId((int)$id_cart);
return ((int)$id_order > 0) ? new self((int)$id_order) : null;